File: examples/cpp/service.cpp
#include <iostream>
#include <string>
#include <cstring>
#include <cstdlib>
#include <openssl/md5.h>
#include <mysql/mysql.h>

class DataService {
private:
    MYSQL* connection;
    
public:
    void getUserInfo(const std::string& userId, const std::string& currentUser) {
        // No authorization check
        fetchUser(userId);
    }
    
    std::string hashPassword(const std::string& password) {
        // Weak hashing
        unsigned char digest[MD5_DIGEST_LENGTH];
        MD5((unsigned char*)password.c_str(), password.length(), digest);
        
        char mdString[33];
        for(int i = 0; i < 16; i++) {
            sprintf(&mdString[i*2], "%02x", (unsigned int)digest[i]);
        }
        return std::string(mdString);
    }
    
    void findUser(const std::string& username) {
        // SQL injection vulnerability
        std::string query = "SELECT * FROM users WHERE username = '" + username + "'";
        mysql_query(connection, query.c_str());
    }
    
    void processInput(const char* input) {
        char buffer[100];
        // Buffer overflow vulnerability
        strcpy(buffer, input);
        executeLogic(buffer);
    }
    
    void initialize() {
        // Hardcoded credentials
        const char* apiKey = "sk-1234567890abcdef";
        std::cout << "Debug mode enabled" << std::endl;
    }
    
    bool authenticate(const std::string& username, const std::string& password) {
        // Weak authentication
        return username == "root" && password.length() > 7;
    }
    
    void loadConfig(const char* data) {
        // Unsafe operations
        char* config = (char*)malloc(strlen(data) + 1);
        strcpy(config, data);
        // Memory not freed - memory leak
    }
    
    void logActivity(const std::string& username, const std::string& token) {
        // Logging sensitive data
        std::cout << "User " << username << " token: " << token << std::endl;
    }
    
    void fetchURL(const std::string& url) {
        // Command injection via system call
        std::string command = "curl " + url;
        system(command.c_str());
    }
    
    void runCommand(const std::string& cmd) {
        // Direct command execution
        system(cmd.c_str());
    }
    
private:
    void fetchUser(const std::string& userId) {}
    void executeLogic(const char* input) {}
};

File: examples/rust/service.rs
use std::process::Command;
use std::collections::HashMap;
use md5;

pub struct DataService {
    db_connection: String,
}

impl DataService {
    pub fn get_user_info(&self, user_id: &str, current_user: &str) {
        // No authorization check
        self.fetch_user(user_id);
    }
    
    pub fn hash_password(&self, password: &str) -> String {
        // Weak hashing algorithm
        format!("{:x}", md5::compute(password))
    }
    
    pub fn find_user(&self, username: &str) {
        // SQL injection vulnerability (simulated)
        let query = format!("SELECT * FROM users WHERE username = '{}'", username);
        self.execute_query(&query);
    }
    
    pub fn process_input(&self, input: &str) {
        // No input validation
        self.execute_logic(input);
    }
    
    pub fn initialize(&self) {
        // Hardcoded secrets
        let api_key = "sk-1234567890abcdef";
        std::env::set_var("DEBUG", "true");
        println!("API Key: {}", api_key);
    }
    
    pub fn authenticate(&self, username: &str, password: &str) -> bool {
        // Weak authentication
        username == "root" && password.len() > 8
    }
    
    pub fn load_config(&self, data: &str) -> HashMap<String, String> {
        // Unsafe deserialization simulation
        let mut config = HashMap::new();
        config.insert("data".to_string(), data.to_string());
        config
    }
    
    pub fn log_activity(&self, username: &str, token: &str) {
        // Logging sensitive information
        println!("User {} access token: {}", username, token);
    }
    
    pub fn fetch_url(&self, url: &str) {
        // SSRF vulnerability
        let output = Command::new("curl")
            .arg(url)
            .output()
            .expect("Failed to execute command");
    }
    
    pub fn run_command(&self, cmd: &str) {
        // Command injection
        Command::new("sh")
            .arg("-c")
            .arg(cmd)
            .output()
            .expect("Failed to execute command");
    }
    
    fn fetch_user(&self, user_id: &str) {}
    fn execute_query(&self, query: &str) {}
    fn execute_logic(&self, input: &str) {}
}